council/llm/llm_middleware.py

```diff
@@ -120,6 +120,41 @@ def __call__(self, llm: LLMBase, execute: ExecuteLLMRequest, request: LLMRequest
         return response
 
 
+class LLMFileLoggingMiddleware:
+    """Middleware for logging LLM requests and responses into a file."""
+
+    def __init__(self, log_file: str, component_name: str) -> None:
+        """Initialize the middleware with the path to the log_file."""
+
+        self.log_file = log_file
+        self.component_name = component_name
+
+    def __call__(self, llm: LLMBase, execute: ExecuteLLMRequest, request: LLMRequest) -> LLMResponse:
+        self._log_llm_request(request)
+        response = execute(request)
+        self._log_llm_response(response)
+        return response
+
+    def _log_llm_request(self, request: LLMRequest) -> None:
+        messages_str = "\n\n".join(message.format() for message in request.messages)
+        self._log(f"LLM input for {self.component_name}:\n{messages_str}")
+
+    def _log_llm_response(self, response: LLMResponse) -> None:
+        if response.result is None:
+            self._log(f"LLM output for {self.component_name} is not available")
+            return
+        self._log(
+            f"LLM output for {self.component_name} Duration: {response.duration:.2f} Output:\n"
+            f"{response.result.first_choice}"
+        )
+
+    def _log(self, content: str) -> None:
```

Comment:
Good for now and probably enough for testing, but I want to point out that this may not behave properly in context of multi-thread. 
I'd suggest adding a `lock` for the `_log` function